src/web/mjs/connectors/kuaikanmanhua.mjs

```diff
@@ -20,82 +20,70 @@ export default class Kuaikanmanhua extends Connector {
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchDOM(request, 'div.tagContent div a', 5);
+        return data.map(element => {
+            return {
+                id: this.getRootRelativeOrAbsoluteLink(element, this.url),
+                title: element.text.trim()
+            };
+        });
     }
 
-    async _getChapterList(manga, callback) {
-        try {
-            let request = new Request(this.url + manga.id, this.requestOptions);
-            let data = await this.fetchDOM(request, 'div.TopicList div div.TopicItem div.title a');
-            let chapterList = data.map(element => {
-                return {
-                    id: this.getRootRelativeOrAbsoluteLink(element, request.url),
-                    title: element.text.trim(),
-                    language: ''
-                };
-            }).filter(chapter => chapter.id !== 'javascript:void(0);');
-            callback(null, chapterList);
-        } catch (error) {
-            console.error(error, manga);
-            callback(error, undefined);
+    async _getChapters(manga) {
+        let chapterList = await this.resolveChapterList(manga, 'comics');
+        if (chapterList.length < 1) {
+            chapterList = await this.resolveChapterList(manga, 'comicList');
         }
+        if (chapterList.length < 1) {
+            console.error('No chapters found !', this);
+        }
+        return chapterList;
     }
 
-    async _getPageList(manga, chapter, callback) {
-        let newRequestOptions = Object.assign({}, this.requestOptions);
-        newRequestOptions.headers.set(
-            'x-user-agent',
-            'Mozilla/5.0 (iPhone; CPU iPhone OS 11_0 like Mac OS X) AppleWebKit/604.1.38 (KHTML, like Gecko) Version/11.0 Mobile/15A372 Safari/604.1'
-        );
-
-        try {
-            let script = `
+    async resolveChapterList(manga, objname) {
+        let chapterList = [];
+        const script = `
             new Promise(resolve => {
-                let pages = __NUXT__.data[0].comicInfo.comic_images.map(img => img.url);
+                let pages = __NUXT__.data[0].`+objname+`.map( comic => {
+                    return {
+                        id : '/web/comic/'+comic.id,
+                        title: comic.title.trim()
+                    };
+                }).reverse();
                 resolve(pages);
-            });`;
-            let request = new Request(this.url + chapter.id, newRequestOptions);
-            let pageList = await Engine.Request.fetchUI(request, script);
-            callback(null, pageList);
-        } catch (error) {
-            console.error(error, manga);
-            callback(error, undefined);
-        }
+            });
+            `;
+        let request = new Request(this.url + manga.id, this.requestOptions);
+        chapterList = await Engine.Request.fetchUI(request, script);
+
+        return chapterList;
```

Comment:
unneeded variable `chapterList `? (the initial [] assignment is overwritten)
`return await Engine.Request.fetchUI(request, script);`